contracts/autonomous-admin/DecentAutonomousAdminV1.sol

```diff
@@ -1,14 +1,14 @@
 // SPDX-License-Identifier: MIT
 pragma solidity 0.8.28;
 
-import {IHats} from "./interfaces/hats/full/IHats.sol";
-import {IHatsElectionsEligibility} from "./interfaces/hats/full/modules/IHatsElectionsEligibility.sol";
+import {IHats} from "../interfaces/hats/IHats.sol";
+import {IHatsElectionsEligibility} from "../interfaces/hats/modules/IHatsElectionsEligibility.sol";
 import {FactoryFriendly} from "@gnosis.pm/zodiac/contracts/factory/FactoryFriendly.sol";
 import {ERC165} from "@openzeppelin/contracts/utils/introspection/ERC165.sol";
-import {IDecentAutonomousAdmin} from "./interfaces/IDecentAutonomousAdmin.sol";
+import {IDecentAutonomousAdminV1} from "../interfaces/autonomous-admin/IDecentAutonomousAdminV1.sol";
 
-contract DecentAutonomousAdmin is
-    IDecentAutonomousAdmin,
+contract DecentAutonomousAdminV1 is
```

Comment:
Maybe also adding `VERSION` constant? So that we can decide which data to run on fly depending from version returned from the contract vs comparing mastercopies? That sounds like best-practice, at least I've seen `Safe` contracts are following that approach

Reply:
That's what the ERC165 implementation is for.

But I suppose that determining which version a given contract address is, using ERC165, is `O(n)` time (`n` being total number of deployed version), but simply asking for the version number is `O(1)` time.

So yeah, I do like that we're implementing ERC165, but for our specific client's use case (in which we _do know_ that these contracts will _definitely_ implement a public function like `version()`), that makes sense for efficiency's sake.

Reply:
To that end, let me make a `DecentAutonomousAdminBase` interface or abstract contract or something that will require this to be implement.

Reply:
Good call sir

Reply:
pushed